Modules/Loadable/Segmentations/Widgets/qMRMLSegmentationGeometryWidget.cxx

```diff
@@ -0,0 +1,803 @@
+
+  /// Fill geometry info section from geometry image data containing geometry information \sa GeometryImageData
+  void updateGeometryWidgets();
+
+  /// Reset geometry image data \sa GeometryImageData
+  void resetGeometryImageData();
+
+  /// Determine if source node is a non-empty segmentation with a labelmap master
+  bool isSourceASegmentationWithBinaryLabelmapMaster();
+
+  /// Determine if input segmentation is non-empty, and with binary labelmap master
+  bool inputSegmentationCanBeResampled();
+
+  // Match the axes of the input labelmap to the axes of the transformed source node.
+  // Use this calculated permutation for updating spacing widget from geometry and interpreting spacing input
+  void matchInputAndSourceAxes(vtkMRMLNode* sourceNode);
+
+public:
+  /// Segmentation MRML node
+  vtkWeakPointer<vtkMRMLSegmentationNode> SegmentationNode;
+
+  /// Oriented image data containing output geometry
+  vtkOrientedImageData* GeometryImageData;
+
+  /// Source to input axes mapping \sa matchInputAndSourceAxes
+  int InputAxisIndexForSourceAxis[3];
+  /// Inverse permutation of \sa InputAxisIndexForSourceAxis
+  int SourceAxisIndexForInputAxis[3];
+
+  /// Flag indicating whether editing is enabled in the widget
+  bool EditEnabled;
+};
+
+//-----------------------------------------------------------------------------
+qMRMLSegmentationGeometryWidgetPrivate::qMRMLSegmentationGeometryWidgetPrivate(qMRMLSegmentationGeometryWidget& object)
+  : q_ptr(&object)
+  , InputAxisIndexForSourceAxis{ 0, 1, 2 }
+  , SourceAxisIndexForInputAxis{ 0, 1, 2 }
+  , EditEnabled(false)
+{
+  this->SegmentationNode = nullptr;
+
+  this->GeometryImageData = vtkOrientedImageData::New();
+}
+
+//-----------------------------------------------------------------------------
+qMRMLSegmentationGeometryWidgetPrivate::~qMRMLSegmentationGeometryWidgetPrivate()
+{
+  if (this->GeometryImageData)
+    {
+    this->GeometryImageData->Delete();
+    this->GeometryImageData = nullptr;
+    }
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLSegmentationGeometryWidgetPrivate::init()
+{
+  Q_Q(qMRMLSegmentationGeometryWidget);
+  this->setupUi(q);
+
+  // Make connections
+  QObject::connect(this->MRMLNodeComboBox_SourceGeometryNode, SIGNAL(currentNodeChanged(vtkMRMLNode*)),
+    q, SLOT(onSourceNodeChanged(vtkMRMLNode*)) );
+  QObject::connect(this->DoubleSpinBox_OversamplingFactor, SIGNAL(valueChanged(double)),
+    q, SLOT(updateWidgetFromMRML()) );
+  QObject::connect(this->checkBox_IsotropicSpacing, SIGNAL(toggled(bool)),
+    q, SLOT(updateWidgetFromMRML()) );
+  QObject::connect(this->MRMLCoordinatesWidget_Spacing, SIGNAL(coordinatesChanged(double*)),
+    q, SLOT(updateWidgetFromMRML()) );
+
+  q->setEnabled(this->SegmentationNode.GetPointer());
+}
+
+//-----------------------------------------------------------------------------
+QString qMRMLSegmentationGeometryWidgetPrivate::calculateOutputGeometry()
```

Comment:
It would be nice to make this available from a logic-type class (so that you can set segmentation geometry without instantiating this widget). If it's difficult to do, then it is not a blocking issue, it can be done in the later, when someone actually needs this.